crates/amaru-kernel/src/lib.rs

```diff
@@ -213,6 +214,30 @@ impl RedeemersExt for Redeemers {
     }
 }
 
+#[derive(Eq, PartialEq)]
+pub struct ScriptRefWithHash {
+    pub hash: ScriptHash,
+    pub script: ScriptRef,
+}
```

Comment:
Debatable. I see room for something more holistic here. `WithHash<A>` or `Addressable<A>`. Theoretically, Pallas should also provides the API making this entirely superfluous.

Reply:
Would it make sense to create a pallas issue to track this?